core/chains/evm/client/pool.go

```diff
@@ -195,36 +225,23 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
-	nodes := p.liveNodes()
-	nNodes := len(nodes)
-	if nNodes == 0 {
+func (p *Pool) selectNode() Node {
+	node := p.nodeSelector.Select()
+
+	if node == nil {
 		p.logger.Critical("No live RPC nodes available")
```

Comment:
Maybe we want to add the selector name to the log message?

Reply:
Done